rdtools/test/normalization_pvwatts_test.py

```diff
@@ -83,6 +83,12 @@ def test_normalization_with_pvw(self):
         # Test corrected energy is equal to 1.0
         self.assertTrue((corr_energy == 1.0).all())
 
+        # Test expected behavior when energy has no explicit frequency
```

Comment:
Great to test this. This test does not cover the case where freq is None and energy time series is not inferable or irregular. As mentioned above, a `with self.assertRaises(ValueError):` or `with pytest.raises(ValueError):` context would do the trick.

Reply:
I've added the test for both pvwatts and sapm functions